lib/Intlc/Linter.hs

```diff
@@ -0,0 +1,37 @@
+module Intlc.Linter where
+
+import           Intlc.ICU
+import           Prelude   hiding (Type)
+
+data Status
+  = Success
+  | Failure Text
+  deriving (Eq, Show)
+
+countInterpolations :: NEStream -> Int
+countInterpolations = foldr go 0
+  where
+    count = \case
```

Comment:
I count three pattern matches with `case` that could be inlined into their functions. :stuck_out_tongue: e.g.

```hs
-- before
f x = case x of y -> z
-- or
f = \case y -> z

-- after
f y = z
```

Reply:
I've found these two helpful to change 5cc966614b0baedadedbf40df9c82b5e68214b2f

I'm guessing the third one is `count = \case` although if the equivalent would be

```haskell
count String = 0
count Number = 0
count (Bool {}) = 1
```

Would this be very similar to using point-free `case` here? Or perhaps there is another equivalent I'm not seeing 👍

Reply:
Yep that's the one, and yeah it's very similar, just a case of common idioms. :slightly_smiling_face: